jena/src/main/java/com/inrupt/client/jena/JenaBodyHandlers.java

```diff
@@ -65,12 +83,39 @@ public static Response.BodyHandler<Model> ofModel() {
     }
 
     /**
-     * Populate a Jena {@link Graph} with an HTTP response.
+     * Populate a Jena {@link Model} with an HTTP response body.
      *
      * @return an HTTP body handler
+     * @deprecated Use ofJenaModel instead for consistent HTTP error handling.
      */
-    public static Response.BodyHandler<Graph> ofGraph() {
-        return responseInfo -> responseInfo.headers().firstValue(CONTENT_TYPE)
+    public static Response.BodyHandler<Model> ofModel() {
+        return JenaBodyHandlers::responseToModel;
+    }
+
+    /**
+     * Populate a Jena {@link Model} with an HTTP response body.
+     *
+     * @return an HTTP body handler
+     */
+    public static Response.BodyHandler<Model> ofJenaModel() {
```

Comment:
Looking at this I think I have an idea of what's wrong with the new module dependency criticized above.

I'll try to argue that the body handler is not the right place to translate a response into problem details.
Whoever is invoking the body handler (our code, not users') should know whether the response is 'ok' (and thus eligible for handling) or 'not ok' (thus eligible for problem detail parsing instead).

I haven't exercised this theory in the context of the codebase, but it seems sound.

Reply:
As we discussed yesterday, the body handler is provided to the HTTP client, at which layer there is no business logic about what is an error vs what isn't. I think this means there is no other place to handle error responses but the body handler itself.